src/controllers/fetch_email.js

```diff
@@ -315,10 +336,164 @@ export class FetchController extends BaseAPIController {
                         }
                     });
                 });
-
             })
             .catch(this.handleErrorResponse.bind(null, res));
     }
+
+    mailAttachment = (req, res, next) => {
+        req.email.findOne({ _id: req.params.mongo_id }, (error, data) => {
+            if (error) {
+                next(new Error(error));
+            } else {
+                if (data) {
+                    let sender_mail = data.get("sender_mail");
+                    let to = data.get("to");
+                    let uid = data.get("uid");
+                    this._db.Imap.findOne({ email: to })
+                        .then((data) => {
+                            var imap = new Imap({
```

Comment:
this code has been used in many places

i.e find active imap setting and start connection with image. 

i have seen same code copy pasted 3 times already.

make a single service for all imap codes